src/nxdoc/nuxeo-server/platform-services/comments.md

```diff
@@ -20,7 +20,73 @@ Comments (and so annotations) are linked to a specific document (their parent).
 
 ## Core Implementation
 
-### Storage
+### Current implementation - TreeCommentManager
+
+#### Storage
+
+Comments in Nuxeo are stored as regular documents of type `Comment`.</br>
+Comments hold:
+
+- parent id (legacy purpose)
+- ancestor ids (legacy purpose)
+- author
+- text
+- creation date
+- modification date
+
+[`Comment`](http://community.nuxeo.com/api/nuxeo/latest/javadoc/org/nuxeo/ecm/platform/comment/api/Comment.html) interface has been created to handle comments in Nuxeo. The interface is used as based type in [comment service](#java-api) and also defines how comments are serialized in JSON in order to use them with the [REST API](#json-entity-format).
+
+Annotations are stored as regular document of type `Annotation` which inherit from `Comment`. They hold in addition of comments:
+
+- blob xpath
+
+Like comments, annotations have the [`Annotation`](http://community.nuxeo.com/api/nuxeo/latest/javadoc/org/nuxeo/ecm/platform/comment/api/Annotation.html) interface to use them.
+
+Comments and annotations can have the facet `ExternalEntity`. When they have this facet, it can hold in addition:
+
+- external entity id
+- external entity
+- external entity origin</br>
+With this facet, it is possible to handle external content such as annotations made in [ARender]({{page space='nxdoc' page='nuxeo-arender-connector'}}) for instance.
+
+Comments and annotations are stored as a tree structure in a folderish document of type `CommentRoot` which is located right under the commented document. Comments are _special children_ which means they are not copied with the commented document during copy operations but they are snapshot when the document is checked in.</br>
+Please note comments are not restored when a version of the commented document is checked out.
```

Comment:
I think we can add more explanation as each comment is stored under the commented document (which it means in the case of reply it will be under the comment). as reminder the Tree API comment allows the tree structure but the web-ui will send always the first comment as a parent which means even we allows the tree we can store them as a flat under the first comment (like Slack, Kevin discussion :) )

Reply:
I can add the following *beautifull* ascii schema
doc
|_CommentRoot
    |_comment
        |_reply

Reply:
looks good to me too the schema. my main wondering is we propos this structure but it's not used in web-ui which is the main client for this API. let wait for the team review ;)